03_python_fastapi_project/main.py

```
@@ -0,0 +1,119 @@
+        raise HTTPException(status_code=404, detail="Product not found")
+    
+    return product
+
+@app.get("/products", response_model=List[ProductResponse])
+async def get_products(db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(Product))
+    products = result.scalars().all()
+    return products
+
+@app.post("/products", response_model=ProductResponse)
+async def create_product(product: ProductDTO, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)